Sources/SwiftlyCore/HTTPClient.swift

```diff
@@ -68,14 +163,42 @@ public struct SwiftlyHTTPClient {
 
     /// Return an array of released Swift versions that match the given filter, up to the provided
     /// limit (default unlimited).
-    ///
-    /// TODO: retrieve these directly from swift.org instead of through GitHub.
     public func getReleaseToolchains(
+        platform: PlatformDefinition,
+        arch a: String? = nil,
         limit: Int? = nil,
         filter: ((ToolchainVersion.StableRelease) -> Bool)? = nil
     ) async throws -> [ToolchainVersion.StableRelease] {
-        let filterMap = { (gh: GitHubTag) -> ToolchainVersion.StableRelease? in
-            guard let release = try gh.parseStableRelease() else {
+        let arch = if a == nil {
+#if arch(x86_64)
+            "x86_64"
+#elseif arch(arm64)
+            "aarch64"
+#else
+            #error("Unsupported processor architecture")
+#endif
```

Comment:
we do this in a few places, can we extract this to a method/computed variable on `Platform`? I think it would also be good to define an enum for it so we don't have to rely on magic strings here and elsewhere.

Reply:
This pattern is isolated to two places at the moment, mostly isolated to path/filename calculation and internal swift org metadata. It might be possible to isolate all of it in the HTTPClient, or an HTTPClient+SwiftOrg.

There are some problems with pulling this out into a global enum at the Platform level. In macOS, the arch isn't used at all because of the universal binaries/packages. Also, there is some disagreement between the naming of the 64-bit arm architecture. In some places. it's "arm64" while others is "aarch64." It doesn't feel like an enum could be universally defined across OSes and cases without being a leaky abstraction.

I feel that in Swift it's quite common to put these kinds of macros (os/arch) at the place where they are needed as long as it isn't excessive and it can maintain locality of change when/if the logic changes. For now, it's only these two places: HTTPClient, and Install.